src/stringobj.h

```diff
@@ -281,10 +317,29 @@ extern Int IsStringConv (
 void MakeImmutableString(Obj str);
 
 
-Obj MakeString(const Char *cstr);
+// Functions to create mutable and immutable GAP strings from C strings.
+// MakeString and MakeImmString are inlineable so 'strlen' can be optimised
+// away for constant strings.
+
+static inline Obj MakeString(const Char * cstr)
+{
+    Obj result;
+    C_NEW_STRING(result, strlen(cstr), cstr);
+    return result;
+}
+
+static inline Obj MakeImmString(const Char * cstr)
+{
+    Obj result = MakeString(cstr);
+    C_NEW_STRING(result, strlen(cstr), cstr);
+    RetypeBag(result, IMMUTABLE_TNUM(T_STRING));
+    return result;
+}
+
+
 Obj MakeString2(const Char *cstr1, const Char *cstr2);
 Obj MakeString3(const Char *cstr1, const Char *cstr2, const Char *cstr3);
-Obj MakeImmString(const Char *cstr);
+
 Obj MakeImmString2(const Char *cstr1, const Char *cstr2);
 Obj MakeImmString3(const Char *cstr1, const Char *cstr2, const Char *cstr3);
 Obj ConvImmString(Obj str);
```

Comment:
Are the `Make(Imm)String{2,3}` variants actually *used* by anyone anywhere? Maybe my grepping-foo is not strong enough, but if noone uses them we should remove them while we're here.